src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -485,11 +487,28 @@ Loader.prototype._nodeModuleNameToPath = function(currPath, moduleName) {
   }
 
   var resolveError = null;
+  var paths = [];
   try {
-    return resolve.sync(moduleName, {
-      basedir: path.dirname(currPath),
-      extensions: ['.js', '.json']
-    });
+    if (NODE_PATH) {
+      if (NODE_PATH.indexOf(process.cwd()) !== -1) {
+        paths.push(NODE_PATH);
+      }
+      else {
+        paths.push(process.cwd() + '/' + NODE_PATH);
+      }
+
+      return resolve.sync(moduleName, {
+        paths: paths,
+        basedir: path.dirname(currPath),
+        extensions: ['.js', '.json']
```

Comment:
Looks like this list of extensions has been moved into `config.moduleFileExtensions` now:

https://github.com/facebook/jest/blob/master/src/HasteModuleLoader/HasteModuleLoader.js#L489